### ISSUE TITLE
 Stuttering with Max98357a

### ISSUE THREAD

**Issue by JelleVE** [pschatzmann/ESP32-A2DP#370]
I am using the max98357a as an external DAC. When I try to stream music to the esp32 (with my iPhone X), it sounds recognizable, but it's stuttering very badly. I have tried a number of different configurations, all with either severe stuttering (often at a high seemingly fixed frequency (3-4 times per second)) as a result or, sometimes, delayed robotic sounds.

I have used this setup before with the [arduino-audio-tools](https://github.com/pschatzmann/arduino-audio-tools) library and there it works flawlessly. In both cases the esp32 was connected to my pc through usb. The bluetooth connection itself seems very stable.

Any ideas as to how to resolve this are much appreciated!

```
#include "BluetoothA2DPSink.h"

BluetoothA2DPSink a2dp_sink;

void setup() {
    static const i2s_config_t i2s_config = {
         .mode = (i2s_mode_t) (I2S_MODE_MASTER | I2S_MODE_TX),
         .sample_rate = 44100,
         .bits_per_sample = I2S_BITS_PER_SAMPLE_16BIT,
         .channel_format = I2S_CHANNEL_FMT_RIGHT_LEFT,
         .communication_format = (i2s_comm_format_t) (I2S_COMM_FORMAT_I2S_MSB | I2S_COMM_FORMAT_I2S_LSB),
         .intr_alloc_flags = 0, // default interrupt priority
         .dma_buf_count = 8,
         .dma_buf_len = 64,
         .use_apll = false,
         .tx_desc_auto_clear = true // avoiding noise in case of data unavailability
    };

    a2dp_sink.set_channels (I2S_CHANNEL_MONO);
    a2dp_sink.set_i2s_config(i2s_config);
    a2dp_sink.set_auto_reconnect(true, 1000);
    a2dp_sink.start("MyMusic");  
}

void loop() {
}
```

**Comment by pschatzmann** [comment]
Other users had this kind of effect when there were some objects between the bluetooth sender and receiver or the ESP32 antenna was damaged. Did you try with another esp32 device or another bluetooth audio source ?

**Comment by JelleVE** [comment]
Thank you for your comment. I have tried with two other esp32 devices (all three of them ESP32 NODEMCU Module WiFi Development Board with CP2102). With both I can play internet radio using the arduino-audio-tools library. I can connect to both using Bluetooth. One has the same issues as described above. The other doesn't play any audio over BT at all. When using a different audio source the problem remains.

This might be out of scope for this repo, but is there any way to check if the antenna was damaged?

**Comment by pschatzmann** [comment]
Not sure, since myself I never had this problem. 
I have no hopes, but you could try to do the output via the audiotools: https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-basic-api/basic-a2dp-i2s/basic-a2dp-i2s.ino

**Comment by JelleVE** [comment]
I have done some more tests: it very much seems Bluetooth related. If I put my phone next to the ESP32 (touching), I now get a much better result (regardless of the library used). I have the impression that quite a bit of data is still getting lost: the music is no longer stuttering, but rather skipping (but it is very, very recognizable). Also, when I use the volume button on my phone, only every so often it registers with the ESP32 (yesterday this did not work at all).

I tried activating use_apll and used different buffer sizes, but the effect felt minimal.

I have again tried with a second source device, and the behavior is identical. Given the fact that the proximity is such a dominant factor (touching vs 5cm distance is a huge difference), it would make sense that the BT antenna was damaged, though it seems somewhat unlikely that that would be the case for three ESPs. I have also noticed that I can only connect a single device with the ESP (only once I disconnect one, I can connect the other). I'm pretty sure this is impossible, but do you think it likely that my ESP32 is functioning in some type of power-savings mode (the specific board implements both Bluetooth classic and BLE)?

**Comment by pschatzmann** [comment]
That's really strange: I can move my Android phone (or my very old IPAD) to the next room w/o any issues. 
I did't do any tests lately, so I tested today again with the latest ESP32 Arduino Release which is 2.0.6 and for me it is still working perfectly..

**Comment by JelleVE** [comment]
I was on 2.0.7 but downgraded to 2.0.6, using ESP32 Dev Module. ([https://raw.githubusercontent.com/espressif/arduino-esp32/gh-pages/package_esp32_index.json](url))

Currently playing flawlessly within a 5cm range. Any further and I'm losing data. Huge improvement though :D

**Comment by pschatzmann** [comment]
Just another thought: I was double checking your Max98357a which might pull a lot of power. Bluetooth is also using a lot of power, so I suggest that you try to power the audio module from an external source.

**Comment by garageeks** [comment]
I agree it must be power constrained. The other day I couldn't explain why it was distorting the sound that much, then I realized I forgot to plug in the power supply and was powered only by USB

**Comment by JelleVE** [comment]
I've now tried using an external power source for the audio module, but unfortunately it didn't help. At this point I am thinking of buying a new esp board, as I have the feeling that I'm running out of options.

**Comment by JelleVE** [comment]
Works now with a new ESP32. 
This one (3 actually) did not work: ESP32 Dev Kit C V2 (NodeMCU).
This one does work: ESP32 Dev KitC V4 ESP32-WROOM-32.

Again, thanks for all the help!
